packages/block-library/src/navigation-menu/edit.js

```diff
@@ -108,14 +113,70 @@ function NavigationMenu( {
 		}
 	};
 
-	useEffect( () => {
-		// Set/Unset colors CSS classes.
-		setAttributes( {
-			backgroundColorCSSClass: backgroundColor.class ? backgroundColor.class : null,
-			textColorCSSClass: textColor.class ? textColor.class : null,
-		} );
-	}, [ backgroundColor.class, textColor.class ] );
+	const handleCreateEmpty = () => {
+		setBlocksTemplate( null );
+		setInitialPlaceholder( false );
+	};
+
+	const handleCreateFromExisting = () => {
+		setBlocksTemplate( defaultPagesMenuItems );
+		setInitialPlaceholder( false );
+	};
+
+	//
+	// MARKUP
+	//
+
+	// Build Inline Styles
+	const navigationMenuInlineStyles = {
+		...( textColor && { color: textColor.color } ),
+		...( backgroundColor && { backgroundColor: backgroundColor.color } ),
+	};
+
+	// Build ClassNames
+	const navigationMenuClasses = classnames(
+		'wp-block-navigation-menu', {
+			'has-text-color': textColor.color,
+			'has-background-color': backgroundColor.color,
+			[ attributes.backgroundColorCSSClass ]: attributes && attributes.backgroundColorCSSClass,
+			[ attributes.textColorCSSClass ]: attributes && attributes.textColorCSSClass,
+		}
+	);
 
+	// UI State: initial placeholder
+	if ( ! hasExistingNavItems && initialPlaceholder ) {
+		return (
+			<Placeholder
+				className="wp-block-navigation-menu-placeholder"
+				icon="menu"
+				label={ __( 'Navigation Menu' ) }
+			>
+				<p className="wp-block-navigation-menu-placeholder__instructions">
+					{ __( 'Create a Menu from all existing pages, or create an empty one.' ) }
+				</p>
+
+				<div className="wp-block-navigation-menu-placeholder__buttons">
+					<Button
+						className="wp-block-navigation-menu-placeholder__button"
+						isDefault={ true }
```

Comment:
Minor tip: JSX allows `<Foo isDefault />` as shorthand for `<Foo isDefault={ true } />`.

Reply:
Noted.